tests/impl.cpp

```diff
@@ -8267,25 +8267,46 @@ result_t test_mm_cvtepu8_epi64(const SSE2NEONTestImpl &impl, uint32_t iter)
     return validateInt64(ret, i0, i1);
 }
 
-result_t test_mm_dp_pd(const SSE2NEONTestImpl &impl, uint32_t iter)
-{
-    const double *_a = (const double *) impl.mTestFloatPointer1;
-    const double *_b = (const double *) impl.mTestFloatPointer2;
-    // FIXME: Different immediate value should be tested
-    const int imm = 0xFF;
-    double d[2];
-    double sum = 0;
-
-    for (size_t i = 0; i < 2; i++)
-        sum += ((imm) & (1 << (i + 4))) ? _a[i] * _b[i] : 0;
-    for (size_t i = 0; i < 2; i++)
-        d[i] = (imm & (1 << i)) ? sum : 0;
+#define ENUM_MM_DP_PD_TEST_CASES(imm8)                               \
+    do {                                                             \
+        const double *_a = (const double *) impl.mTestFloatPointer1; \
+        const double *_b = (const double *) impl.mTestFloatPointer2; \
+        const int imm = imm8;                                        \
+        double d[2];                                                 \
+        double sum = 0;                                              \
+        for (size_t i = 0; i < 2; i++)                               \
+            sum += ((imm) & (1 << (i + 4))) ? _a[i] * _b[i] : 0;     \
+        for (size_t i = 0; i < 2; i++)                               \
+            d[i] = (imm & (1 << i)) ? sum : 0;                       \
+        __m128d a = load_m128d(_a);                                  \
+        __m128d b = load_m128d(_b);                                  \
+        __m128d ret = _mm_dp_pd(a, b, imm);                          \
+        if (validateDouble(ret, d[0], d[1]) != TEST_SUCCESS)         \
+            return TEST_FAIL;                                        \
+    } while (0);
```

Comment:
Remove the trailing semicolon (`;`) and append it at the end of each occurrence of `ENUM_MM_DP_PD_TEST_CASES`.

Reply:
Done.